out_writeapi.go

```diff
@@ -284,16 +284,22 @@ func sendRequestExactlyOnce(ctx context.Context, data [][]byte, config **outputC
 func sendRequestRetries(ctx context.Context, data [][]byte, config **outputConfig, streamIndex int) error {
 	retryer := newStatelessRetryer((*config).numRetries)
 	attempt := 0
-	currStream := (*(*config).managedStreamSlice)[streamIndex]
 	for {
 		err := sendRequestExactlyOnce(ctx, data, config, streamIndex)
 		if err == nil {
 			break
 		}
 		//unsuccesful data append
 		if rebuildPredicate(err) {
-			currStream.managedstream.Finalize(ctx)
-			currStream.managedstream.Close()
+			streamSlice := (*config).managedStreamSlice
+			// finalizes all streams
+			for i := 0; i < len(*streamSlice); i++ {
+				currStream := (*streamSlice)[i]
+				currStream.managedstream.Finalize(ctx)
+				currStream.managedstream.Close()
+			}
+			// eliminates all streams but the first one to build from the bottom up again
+			(*(*config).managedStreamSlice) = (*(*config).managedStreamSlice)[:1]
```

Comment:
How are the offset and the queue be reset?

Reply:
Done.

Reply:
Resolving this as I assume that this entire section will be revised as part of the other comment about not to close streams before making sure they don't have any requests in queue.

Reply:
Resolving this as I assume that this entire section will be revised as part of the other comment about not to close streams before making sure they don't have any requests in queue.